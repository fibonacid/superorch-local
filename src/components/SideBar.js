import React, {Component} from 'react'
import styled from 'styled-components'

const StyledContainer = styled.div`
  border-right: solid 1px black;
  width: 200px;
  max-width: 300px;
`;

const StyledList = styled.ul`
  overflow-y: auto;
  height: 100%;
`;

const StyledItem = styled.li`
  padding: 5px;
  border-bottom: solid 1px black;
`;


class SideBar extends Component {
  render() {
    return (
      <StyledContainer>
        <StyledList>
          <StyledItem>User 1</StyledItem>
          <StyledItem>User 2</StyledItem>
          <StyledItem>User 3</StyledItem>
        </StyledList>
      </StyledContainer>
    )
  }
}

export default SideBar
