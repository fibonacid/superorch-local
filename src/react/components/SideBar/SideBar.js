import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components/macro";

const StyledContainer = styled.div`
  border-right: solid 1px black;
  width: 200px;
  max-width: 300px;
  overflow-y: auto;
`;

const StyledList = styled.ul`
  height: 100%;
`;

const StyledListItem = styled.li`
  padding: 5px;
  border-bottom: solid 1px black;
`;

function SideBar(props) {
  const { users } = props;
  return (
    <StyledContainer data-test={"SideBarComponent"}>
      <StyledList data-test={"list"}>
        {users.map((user, i) => (
          <StyledListItem data-test={"list-item"} key={i}>
            {user.name} #{user.id}
          </StyledListItem>
        ))}
      </StyledList>
    </StyledContainer>
  );
}

SideBar.propTypes = {
  users: PropTypes.arrayOf(PropTypes.object)
};

export default SideBar;
