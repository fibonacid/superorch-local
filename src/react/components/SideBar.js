import React from 'react'
import * as styled from './styles/SideBar.styles'

function SideBar(props) {
    const {users} = props;
    return (
      <styled.Container
        data-test={'SideBarComponent'} >
        <styled.List
          data-test={'list'} >
          {users.map((user,i) => (
            <styled.ListItem
              data-test={'list-item'}
              key={i}
            >
                {user.name}
            </styled.ListItem>
          ))}
        </styled.List>
      </styled.Container>
    )
}

export default SideBar;
