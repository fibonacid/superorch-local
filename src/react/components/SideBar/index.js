import { connect } from "react-redux";
import SideBar from "./SideBar";
import { selectUsers } from "../../reducers/root";

const mapStateToProps = state => ({
  users: selectUsers(state)
});

export default connect(mapStateToProps, null)(SideBar);
