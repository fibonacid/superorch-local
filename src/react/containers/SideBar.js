import {connect} from "react-redux";
import SideBar from "../components/SideBar";

const mapStateToProps = state => ({
  users: state.chat.users
});

export default connect(mapStateToProps, null)(SideBar);
