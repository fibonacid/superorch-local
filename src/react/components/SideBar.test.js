import React from "react";
import {shallow} from 'enzyme'
import { Provider } from 'react-redux'
import configureStore from "../store";
import SideBar from "./SideBar";

const mockStore = configureStore();

describe("SideBar", () => {
  it ('should render my component', () => {
    // mount component (shallow)
    const component = shallow(
      <Provider store={mockStore}>
        <SideBar/>
      </Provider>
    );
    // unmount component
    component.unmount();
  });
});
