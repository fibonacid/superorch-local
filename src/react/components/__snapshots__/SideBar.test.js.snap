// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideBar Component When has props Should renderer without errors 1`] = `
<SideBarstyles__Container
  data-test="SideBarComponent"
>
  <SideBarstyles__List
    data-test="list"
  >
    <SideBarstyles__ListItem
      data-test="list-item"
      key="0"
    >
      Al
    </SideBarstyles__ListItem>
    <SideBarstyles__ListItem
      data-test="list-item"
      key="1"
    >
      John
    </SideBarstyles__ListItem>
  </SideBarstyles__List>
</SideBarstyles__Container>
`;
